### ISSUE TITLE
FileInput not compatible with PSR (and Dictatoros) UploadedFile

### ISSUE THREAD

**Issue by bfolliot** [zendframework/zend-inputfilter#145]
Hello.

Zend\InputFilter\FileInput does not accept an implementation of Psr\Http\Message\UploadedFileInterface **if the input is required**, for exemple, i have this form with InputFilterProviderInterface :

```php
class MyForm extends Form implements InputFilterProviderInterface
{

    public function __construct()
    {
        parent::__construct();

        $this->add([
            'name' => 'myPicture',
            'type'  => 'File',
        ]);

        $this->add([
            'name' => 'submit',
            'type'  => 'Submit',
            'attributes' => [
                'value' => 'Submit',
                'class' => 'btn-primary',
            ],
        ]);
    }


    /**
      * @return array
      */
    public function getInputFilterSpecification() : array
    {
        return [
            'myPicture' => [
                'required' => true, // <========
            ],
        ];
    }
}
```

And in an Expressive middleware :

```php
$form = new MyForm;

if ($request->getMethod() === 'POST') {
    $form->setData(array_merge_recursive(
        // @var Psr\Http\Message\ServerRequestInterface $request
        $request->getParsedBody(),
        $request->getUploadedFiles()
    ));
    var_dump($form->isValid()); // Display always false...
}

```

**Comment by svycka** [comment]
I workaround-ed this with `zendframework/zend-psr7bridge` like this:
````php
$zendRequest = \Zend\Psr7Bridge\Psr7ServerRequest::toZend($request);
$inputFilter->setData($zendRequest->getFiles());
...
````

**Comment by firefred** [comment]
We have similar problem here, this problem forces us to use $_FILES array instead of $request->getUploadedFiles()

**Comment by froschdesign** [comment]
Fixed and released with [version 2.9.0](https://github.com/zendframework/zend-inputfilter/releases/tag/release-2.9.0).
A code example can be found in the documentation: https://docs.zendframework.com/zend-inputfilter/file-input/#psr-7-support
